joomla/joomla-cms
# [5.0] Remove libraries/vendor/psr/http-message/docs from deleted files and folders lists in script.php

Pull Request for Issue # .

### Summary of Changes

This pull request (PR) removes the following items from the lists of files and folders to be deleted on update in script.php:
- Files `/libraries/vendor/psr/http-message/docs/PSR7-Interfaces.md` and `/libraries/vendor/psr/http-message/docs/PSR7-Usage.md`.
- Folder `/libraries/vendor/psr/http-message/docs`.

The reason why they have been added to the lists before is because these files were present in 4.4-dev full installation packages but not in 5.0-dev full installation packages at the time when the alpha1 releases were prepared.

Meanwhile the files are also present in 5.0-dev full installation packages, so they have to be removed from the lists of files and folders to be deleted on update.

### Request for Comments

Alternatively to this PR or in addition to it, these files and the folder could be removed from building the packages like we do with other build or documentation files from composer or npm dependencies e.g. here: https://github.com/joomla/joomla-cms/blob/5.0-dev/build/build.php#L125-L192 , because they are only documentation of a composer dependency. 

But they are end user documentation and not documentation for maintenance of that library, so maybe they should be kept?

Anyway, if we decide to remove them from the build, this should be done in 4.4-dev and later be merged up to 5.0-dev so that is done in both versions.

I still would say merge this PR so they are removed from the lists of deleted files and folders, and later when they have been removed from the package build in both versions, they will be added back to these lists but at a later section with the correct version comment "// From 5.0.0-alpha1 to 5.0.0-alpha2".

But it also would be ok not to merge this PR here and leave it as it is when the other thing (build.php) is done in 4.4-dev and merged up to 5.0-dev before the next alpha2 releases.

@HLeithner Please check and if necessary discuss with others and decide.

### Testing Instructions

Code review plus check if the files and their parent folder mentioned in the description above are present in following full installation packages:
- [Joomla_4.4.0-alpha1-Alpha-Full_Package.zip](https://github.com/joomla/joomla-cms/releases/download/4.4.0-alpha1/Joomla_4.4.0-alpha1-Alpha-Full_Package.zip)
- [Joomla_5.0.0-alpha1-Alpha-Full_Package.zip](https://github.com/joomla/joomla-cms/releases/download/5.0.0-alpha1/Joomla_5.0.0-alpha1-Alpha-Full_Package.zip)
- Nightly build [Joomla_5.0.0-alpha2-dev-Development-Full_Package.zip](https://developer.joomla.org/nightlies/Joomla_5.0.0-alpha2-dev-Development-Full_Package.zip)

### Actual result BEFORE applying this Pull Request

The files and their parent folder are present in the 4.4.0-alpha1 package and in the latest 5.0.0-alpha2-dev nightly build.

In the 5.0.0-alpha1 package they are not present, that's why they were added to the lists of deleted files and folders.

When updating from 4.4 to 5 or from 5.x to 5.y the files and their parent folder will be deleted.

### Expected result AFTER applying this Pull Request

When updating from 4.4 to 5 or from 5.x to 5.y the files and their parent folder will not be deleted.

### Link to documentations
Please select:
